ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/qos-profiles.j2

```diff
@@ -61,6 +61,47 @@ QOS Profile: **{{ profile.name }}**
 {%                 endfor %}
 {%             endif %}
 {%         endif %}
+
+**ECN Configuration**
+
+| TX queue | Type | ECN Enabled | Min Threshol | Max Threshold | Max Mark Probability |
+| -------- | ---- | ----------- | ------------ | ------------- | -------------------- |
+{%         if profile.tx_queues is arista.avd.defined %}
+{%             for tx_queue in profile.tx_queues | arista.avd.natural_sort('id') %}
+{%                 set type = "All" %}
+{%                 if tx_queue.random_detect.ecn.threshold is arista.avd.defined %}
+{%                     set ecn = "True" %}
+{%                 else %}
+{%                     set ecn = "False" %}
+{%                 endif %}
+{%                 set min = tx_queue.random_detect.ecn.threshold.min | arista.avd.default("-") %}
+{%                 set max = tx_queue.random_detect.ecn.threshold.max | arista.avd.default("-") %}
+{%                 set prob = tx_queue.random_detect.ecn.threshold.max_probability | arista.avd.default("-") %}
+{%                 set units = tx_queue.random_detect.ecn.threshold.units | arista.avd.default("") %}
+| {{ tx_queue.id }} | {{ type }} | {{ ecn }} | {{ min }}  {{ units }} | {{ max }} {{ units }} | {{ prob }} |
+{%             endfor %}
+{%         endif %}
+{%         if profile.uc_tx_queues is arista.avd.defined %}
+{%             for uc_tx_queue in profile.uc_tx_queues | arista.avd.natural_sort('id') %}
+{%                 set type = "Unicast" %}
+{%                 if uc_tx_queue.random_detect.ecn.threshold is arista.avd.defined %}
```

Comment:
Detecting on the presence of a dict, is _not_ a good way to detect if it is enabled or not. If it is possible to just configure `random-detect ecn` with nothing else set, we should add an `enabled: <bool>` key under the dict and detect on that here.

Reply:
After going thought the EOS documentation and testing on `DCS-7050DX4-32S-F` I found that `random-detect ecn` on it's own is not a valid command. I am thus marking 'min', 'max' and 'units' as required in the schema, and testing on the presence of `units` instead of using an added `enabled` bool.